tests/storage/localfssd_test.py

```diff
@@ -682,6 +682,52 @@ def test_volume_create_cow_sparse_with_parent(user_domain, local_fallocate):
     assert int(actual["truesize"]) == qemu_info['actual-size']
 
 
+@pytest.mark.xfail(reason='Preallocation for qcow2 not considered')
+def test_volume_create_cow_prealloc(user_domain, local_fallocate):
+    img_uuid = str(uuid.uuid4())
+    vol_uuid = str(uuid.uuid4())
+
+    user_domain.createVolume(
+        imgUUID=img_uuid,
+        capacity=PREALLOCATED_VOL_SIZE,
+        volFormat=sc.COW_FORMAT,
+        preallocate=sc.PREALLOCATED_VOL,
+        diskType='DATA',
+        volUUID=vol_uuid,
+        desc="Test volume",
+        srcImgUUID=sc.BLANK_UUID,
+        srcVolUUID=sc.BLANK_UUID)
+
+    vol = user_domain.produceVolume(img_uuid, vol_uuid)
+
+    path = vol.getVolumePath()
+    qemu_info = qemuimg.info(path)
+
+    verify_volume_file(
+        path=path,
+        format=qemuimg.FORMAT.QCOW2,
+        virtual_size=PREALLOCATED_VOL_SIZE,
+        qemu_info=qemu_info)
+
+    # Actual size is bigger than than preallocated size as it includes
+    # the qcow2 header, which is one cluster size plus the variable sized
+    # tables. We could calculate the value for the the preallocated size in
+    # this check, but checking than the actual size is bigger should suffice.
+    assert qemu_info['actual-size'] > PREALLOCATED_VOL_SIZE
```

Comment:
This will not be correct for qcow2 image since we don't use qemu to do the preallocation. For preallocated qcow2 we can test that

    actual size >= virutal size.

This check may work on some file systems (xfs) and fail on others. qemu-img configure xfs to use extent size of 1m, which may cause it to report bigger actual size. We don't want to assume this for any file system.